src/cdk-experimental/ui-patterns/tree/tree.ts

```diff
@@ -0,0 +1,517 @@
+        .on([Modifier.Ctrl, Modifier.Meta], ' ', () => this.selectionManager.toggle())
+        .on([Modifier.Ctrl, Modifier.Meta], 'Enter', () => this.selectionManager.toggle())
+        .on([Modifier.Ctrl, Modifier.Meta], 'Home', () => this.first())
+        .on([Modifier.Ctrl, Modifier.Meta], 'End', () => this.last())
+        .on([Modifier.Ctrl, Modifier.Meta], 'A', () => {
+          this.selectionManager.toggleAll();
+          this.selectionManager.select(); // Ensure the currect item remains selected.
+        });
+    }
+
+    manager
+      .on(this.expandKey, () => this.expand())
+      .on(this.collapseKey, () => this.collapse())
+      .on(Modifier.Shift, '*', () => this.expandSiblings());
+
+    return manager;
+  });
+
+  /** The pointerdown event manager for the tree. */
+  pointerdown = computed(() => {
+    const manager = new PointerEventManager();
+
+    if (this.multi()) {
+      manager.on(Modifier.Shift, e => this.goto(e, {selectRange: true}));
+    }
+
+    if (!this.multi() && this.followFocus()) {
+      return manager.on(e => this.goto(e, {selectOne: true}));
+    }
+
+    if (!this.multi() && !this.followFocus()) {
+      return manager.on(e => this.goto(e, {toggle: true}));
+    }
+
+    if (this.multi() && this.followFocus()) {
+      return manager
+        .on(e => this.goto(e, {selectOne: true}))
+        .on(Modifier.Ctrl, e => this.goto(e, {toggle: true}));
+    }
+
+    if (this.multi() && !this.followFocus()) {
+      return manager.on(e => this.goto(e, {toggle: true}));
+    }
+
+    return manager;
+  });
+
+  constructor(readonly inputs: TreeInputs<V>) {
+    this.allItems = inputs.allItems;
+    this.focusMode = inputs.focusMode;
+    this.disabled = inputs.disabled;
+    this.activeIndex = inputs.activeIndex;
+    this.skipDisabled = inputs.skipDisabled;
+    this.wrap = inputs.wrap;
+    this.orientation = inputs.orientation;
+    this.textDirection = inputs.textDirection;
+    this.multi = inputs.multi;
+    this.value = inputs.value;
+    this.selectionMode = inputs.selectionMode;
+    this.typeaheadDelay = inputs.typeaheadDelay;
+    this.focusManager = new ListFocus({
+      ...inputs,
+      items: this.visibleItems,
+    });
+    this.navigationManager = new ListNavigation({
+      ...inputs,
+      wrap: computed(() => this.inputs.wrap() && !this.inSelection()),
+      items: this.visibleItems,
+      focusManager: this.focusManager,
+    });
+    this.selectionManager = new ListSelection({
+      ...inputs,
+      items: this.visibleItems,
+      focusManager: this.focusManager,
+    });
+    this.typeaheadManager = new ListTypeahead({
+      ...inputs,
+      items: this.visibleItems,
+      focusManager: this.focusManager,
+    });
+    this.expansionManager = new ListExpansion({
+      multiExpandable: () => true,
+      // TODO(ok7sai): allow pre-expanded tree items.
+      expandedIds: signal([]),
+      items: this.children,
+      disabled: this.disabled,
+    });
+  }
+
```

Comment:
We should add a todo to add a `setDefaultState` method for initializing the Tree.